specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -0,0 +1,236 @@
+  }
+
+  @doc("Properties of the Event Broker operation.")
+  model BrokerProperties {
+    @doc("The token used to lock the event.")
+    lockToken: LockToken;
+  }
+
+  @doc("Receive operation details per Cloud Event.")
+  model ReceiveDetails {
+    @doc("The Event Broker details.")
+    brokerProperties: BrokerProperties;
+
+    @doc("Cloud Event details.")
+    event: CloudEventEvent;
+  }
+
+  @doc("Details of the Receive operation response.")
+  model ReceiveResponse {
+    @doc("Array of receive responses, one per cloud event.")
+    value: ReceiveDetails[];
+  }
+   
+  @doc("Failed LockToken information.")
+  model FailedLockToken {
+    @doc("LockToken value")
+    lockToken: LockToken;
+
+    @doc("Error code")
+    errorCode: int32;
+
+    @doc("Description of the error")
+    errorDescription: string; 
+  }
+   
+  @doc("Details of the LockTokens information. This is used for both Acknowledge and Release operation response.")
+  model LockTokensResponse {
+    @doc("Array of LockToken values for failed cloud events.")
+    failedLockTokens: FailedLockToken[];
+
+    @doc("Array of LockToken values for succeeded cloud events.")
+    succeededLockTokens: string[];
+  }
+
+  @doc("Lock token input formatting.")
+  model LockTokenInput {
+    @doc("LockTokens")
+    lockTokens: string[];
+  }
+
+  // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}:publish?api-version={apiVersion}}
+
+  @doc("Publish Single Cloud Event to namespace topic.")
+  @route("/topics/{topicName}:publish", {shared: true})
+  @post op PublishCloudEvent is Azure.Core.RpcOperation<{
+    @doc("content type")
+    @header("content-type")
+    contentType: "application/cloudevents+json; charset=utf-8";
+
+    @doc("Topic Name.")
+    @path
+    topicName: string;
+
+    @doc("Single Cloud Event being published.")
+    @body 
+    event: CloudEventEvent;
+  }, OkResponse>;
+
+
+  @doc("Publish Batch of Cloud Events to namespace topic.")
+  @route("/topics/{topicName}:publish", {shared: true})
+  @post op PublishBatchOfCloudEvents is Azure.Core.RpcOperation<{
+    @doc("content type")
+    @header("content-type")
+    contentType: "application/cloudevents-batch+json; charset=utf-8";
+
+    @doc("Topic Name.")
+    @path
+    topicName: string;
+
+    @doc("Array of Cloud Events being published.")
+    @body
+    events: CloudEventEvent[];
+  }, OkResponse>;
+   
+  @doc("Receive Batch of Cloud Events from the Event Subscription.")
+  @route("/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive")
+  @post op ReceiveBatchOfCloudEvents is Azure.Core.RpcOperation<{
+    @doc("Topic Name.")
+    @path
+    topicName: string;
+
+    @doc("Event Subscription Name.")
+    @path
+    eventSubscriptionName: string;
+
+    @doc("Max Events count to be received.")
+    @query
+    maxEvents?: int32 = 1;
+
+    @doc("Timeout value for receive operation in Seconds. Default is 60 seconds.")
+    @query
+    timeout?: int32 =60;
```

Comment:
I think we wanted to name this as maxWaitTime - is there a way to do that given this is query param sent to the service?

Reply:
Is there some format we can apply so this generates as a TimeSpan,Duration,etc? In swagger, we would use format: date-time.

Reply:
Cool, is there an equivalent for `format: date-time`?

Reply:
Type it as `duration`. We're still working on specifiying that the serialization format is an int, so let's keep int for now short term

Reply:
> I think we wanted to name this as maxWaitTime - is there a way to do that given this is query param sent to the service?
```
alias Parameters = {
@query({name:"bad-name"})
CoolName: string;
};

op foo(...Parameters): void;
```